src/main/java/nextstep/laddergame/domain/ladder/Line.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.laddergame.domain.ladder;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class Line {
+    private static final Random random = new Random();
+    private final List<Position> positions;
+
+    public Line(int participantCounts) {
+        this.checkParticipantCounts(participantCounts);
+
+        final List<Position> positionsList = new ArrayList<>();
+        Direction direction = null;
+
+        for (int i = 0; i < participantCounts; i++) {
+            direction = this.makeDirection(direction, i, participantCounts - 1);
+            positionsList.add(new Position(i, direction));
+        }
+
+        this.positions = positionsList;
+    }
+
+    private Direction makeDirection(Direction direction, int currentIndex, int lastIndex) {
+        if (Objects.isNull(direction)) {
+            return Direction.first(random.nextBoolean());
+        }
+
+        if (currentIndex == lastIndex) {
+            return direction.last();
+        }
+
+        return direction.next(random.nextBoolean());
+    }
```

Comment:
> 뭔가 이 방법을 쓰니 List 을 생성하는게 제 코드처럼밖에 안나오는것같습니다.
최상단에 Direction direction = null; 로 direction을 선언하고
갈아치우기 같은느낌인데요
더 좋게 생성하는 방법이 있을지 궁금합니다.

이렇게도 할 수 있을 것 같아요 🤔 

```suggestion
    public Line(int participantCounts) {
        this.checkParticipantCounts(participantCounts);
        for (int i = 0; i < participantCounts; i++) {
            this.positions.add(getPosition(i, random.nextBoolean(), participantCounts - 1));
        }
    }

    private Position getPosition(int index, boolean isPoint, int lastIndex) {
        if (index == 0) {
            return new Position(0, Direction.first(isPoint));
        }

        Position beforePosition = this.positions.get(index - 1);
        if (index == lastIndex) {
            return new Position(index, beforePosition.last());
        }

        return new Position(index, beforePosition.next(isPoint));
    }
```

Reply:
오 이코드를 보고 조금 생각해보니 Position 또한 정적 팩터리 메서드로 관리를 하면 될것같아서
해당 리뷰사항에서 조금 변형하여 적용해봤습니다.